units/src/amount/signed.rs

```diff
@@ -58,6 +58,20 @@ impl SignedAmount {
         SignedAmount::from_float_in(btc, Denomination::Bitcoin)
     }
 
+    /// Converts from a value expressing a whole number of bitcoin to a [`SignedAmount`]
+    /// in const context.
+    ///
+    /// # Panics
+    ///
+    /// The function panics if the argument multiplied by the number of sats
+    /// per bitcoin overflows a `u64` type.
+    pub const fn from_int_btc(btc: u64) -> Amount {
```

Comment:
In e21f325b9d6fe6041d430046c514c8ceb0ab2ec9:

I think we should make this uniform with `from_btc` (which I see is `alloc`-gated, which is a good reason to have this function). That is, it should return a result instead of panicking. (It can return an `OutOfRangeError` rather than a `ParseError` -- the float one returns a `ParseError` because it round-trips through a string to avoid requiring us to think about precision loss when changing bases.)

I see that what you've done here is already implemented in `unsigned.rs`. I think we should change that one too.

I'm also tempted to stick a second `sats: u64` field onto this ... but the value of that seems pretty low. If people want satoshi precision they already have `from_sat`.

Reply:
I added `from_int_btc` and `from_int_btc_const` because `unwrap` doesn't work in const context.

Reply:
It sucks to have an API wart like this because of const fn limitations that are very likely to go away.

So I'm tempted to just live with the `Result` and make users manually unwrap by matching and panicking.